arm-insights/2015-04-01/swagger/insightsClient.json

```diff
@@ -0,0 +1,1123 @@
+      "get": {
+        "tags": [
+          "Metric"
+        ],
+        "operationId": "Metric_Get",
+        "description": "The List Metric operation lists the metric value sets for the resource metrics.",
+        "parameters": [
+          {
+            "name": "resourceUri",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The resource identifier of the target resource to get metrics for.",
+            "x-ms-skip-url-encoding": true
+          },
+          {
+            "name": "$filter",
+            "in": "query",
+            "required": false,
+            "type": "string",
+            "description": "The filter to apply on the operation."
+          },
+          {
+            "name": "api-version",
+            "in": "query",
+            "required": false,
+            "type": "string",
+            "default": "2016-06-01",
+            "description": "The api-version to send to the server"
+          },
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Successful request to get the list of metric values",
+            "schema": {
+              "$ref": "#/definitions/MetricCollection"
+            }
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": null
+        },
+        "x-ms-odata": "#/definitions/Metric"
+      }
+    },
+    "/{resourceUri}/usages": {
+      "get": {
+        "tags": [
+          "UsageMetric"
+        ],
+        "operationId": "UsageMetric_List",
+        "description": "The List operation lists the usage metrics for the resource.",
+        "parameters": [
+          {
+            "name": "resourceUri",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The resource identifier of the target resource to get usages for."
+          },
+          {
+            "name": "api-version",
+            "in": "query",
+            "required": true,
+            "type": "string",
+            "description": "The resource provider api version."
+          },
+          {
+            "name": "$filter",
+            "in": "query",
+            "required": false,
+            "type": "string",
+            "description": "The odata filter string to use in the query"
+          },
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Successful request to get the list of usage metric values",
+            "schema": {
+              "$ref": "#/definitions/UsageMetricCollection"
+            }
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": null
+        },
+        "x-ms-odata": "#/definitions/UsageMetric"
+      }
+    }
+  },
+  "definitions": {
+    "LocalizableString": {
```

Comment:
They could be marked as readOnly, but it won't harm if the user changes the values... no way the user could send this to the server and update history for instance.

How can I mark the properties as readOnly?